### PR TITLE
[FLINK-33008] Refactor to use KubernetesClient from JOSDK Context

### PR DESCRIPTION
**Pull request by darenwkt**
## What is the purpose of the change

Refactor code to use KubernetesClient from JOSDK Context

## Brief change log

KubernetesClient is passed from josdkContext to flinkResourceContext. The rest of the code get client from either context.

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changes to the `CustomResourceDescriptors`: (no)
  - Core observer or reconciler logic that is regularly executed: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not applicable)

### CODE REVIEW

File: flink-kubernetes-operator-api/src/main/java/org/apache/flink/kubernetes/operator/api/listener/FlinkResourceListener.java [hunk 1]
```diff
@@ -45,8 +44,6 @@ public interface FlinkResourceListener extends Plugin {
     interface ResourceContext<R extends AbstractFlinkResource<?, ?>> {
         R getFlinkResource();
 
-        KubernetesClient getKubernetesClient();
```
Comment (gyfora) [inline flink-kubernetes-operator-api/src/main/java/org/apache/flink/kubernetes/operator/api/listener/FlinkResourceListener.java:48]: I think we should not touch this here, I think the FlinkResourceListener interfaces should be reworked but probably not as part of this ticket.

**Review by gyfora — CHANGES_REQUESTED** [review]
Added a minor comment about the resource listener interface, otherwise looks good.

File: flink-kubernetes-operator-api/src/main/java/org/apache/flink/kubernetes/operator/api/listener/FlinkResourceListener.java (hunk 1 above)
Comment (darenwkt) [reply flink-kubernetes-operator-api/src/main/java/org/apache/flink/kubernetes/operator/api/listener/FlinkResourceListener.java]: Thanks for the comment, have addressed it in latest commit

**Comment by darenwkt** [comment]
Thanks Gyula for the review, have addressed the comment in latest commit

**Review by gyfora — APPROVED** [review]

**Pull request merged**